pkg/controllers/user/pipeline/remote/gitlab/webhook.go

```diff
@@ -0,0 +1,96 @@
+package gitlab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/go-querystring/query"
+	"github.com/xanzy/go-gitlab"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+//create webhook,return id of webhook
+func (c *client) createGitlabWebhook(user string, repo string, accesstoken string, webhookURL string, secret string) (string, error) {
+
+	project := url.QueryEscape(user + "/" + repo)
+	client := http.Client{
+		Timeout: 15 * time.Second,
+	}
+	APIURL := fmt.Sprintf(c.API+"/projects/%s/hooks", project)
+	req, err := http.NewRequest("POST", APIURL, nil)
+
+	opt := &gitlab.AddProjectHookOptions{
+		PushEvents:          gitlab.Bool(true),
+		MergeRequestsEvents: gitlab.Bool(true),
+		TagPushEvents:       gitlab.Bool(true),
+		URL:                 gitlab.String(webhookURL),
+		EnableSSLVerification: gitlab.Bool(false),
+		Token: gitlab.String(secret),
+	}
+	q, err := query.Values(opt)
+	if err != nil {
+		return "", err
+	}
+	req.URL.RawQuery = q.Encode()
+	req.Header.Add("Authorization", "Bearer "+accesstoken)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	respData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	if resp.StatusCode > 399 {
```

Comment:
399 is a magic number. Maybe we can use resp.StatusCode >= http.StatusBadRequest?